### PR TITLE
Feat/int 36 toggle group

### PR DESCRIPTION
**Pull request by LimeWub**
## Description
Component that lets users toggle between a group of limited options(Suggested range is 2-6 options).

Functionality based on the [ToggleGroup](https://www.radix-ui.com/docs/primitives/components/toggle-group) radix component, which already allows for: single/multiple select, disabling or partly disabling options, adds keyboard navigation and orientation and more.

Extends visually by allowing for vertical/horizontal display, gaps using stack or joined, rounded corners and so on.

## Screenshots
Screenshot 2022-05-12 at 18 59 22

Screenshot 2022-05-12 at 19 00 15

Screenshot 2022-05-12 at 19 00 32

Screenshot 2022-05-12 at 19 00 52

Screenshot 2022-05-12 at 19 02 09

Screenshot 2022-05-12 at 19 01 22

**Hover:**
Screenshot 2022-05-12 at 19 06 05

**Focus:**
Screenshot 2022-05-12 at 19 05 51

**Selected:**
Screenshot 2022-05-12 at 19 03 51

**Focus + Selected**
Screenshot 2022-05-12 at 19 05 59

**Hover + Selected + Focus**
Screenshot 2022-05-12 at 19 06 25

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -7943,6 +8042,24 @@ mdast-util-from-markdown@0.8.5, mdast-util-from-markdown@^0.8.0:
     parse-entities "^2.0.0"
     unist-util-stringify-position "^2.0.0"
 
+mdast-util-from-markdown@1.2.0:
```
Comment (LimeWub) [inline yarn.lock:8045]: Why is this here! D:
This is so irrelevant to what I've actually added to `package.json`

I've reset and rerun yarn but it's always there - this version was installed as a dependency for the WISIWYG SPIKE. T_T

File: src/components/toggle-group/ToggleGroupItem.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import * as ToggleGroup from '@radix-ui/react-toggle-group'
+
+import { styled } from '~/stitches'
+
+// TODO: How do I make this reusable from other components?
+const focusIndicatorBlock = ({ position = 'relative', zIndex = 1 } = {}) => {
```
Comment (LimeWub) [inline src/components/toggle-group/ToggleGroupItem.tsx:6]: Help w this pls :)

File: src/components/stack/Stack.tsx [hunk 3]
```diff
@@ -34,28 +34,35 @@ const gap = Object.keys(theme.space).reduce(
...
       },
       'row-reverse': {
         flexDirection: 'row-reverse',
-        alignItems: 'center'
       },
       column: {
         flexDirection: 'column'
       }
     },
+    wrap: {
```
Comment (hollg) [inline src/components/stack/Stack.tsx:52]: Could you explain the need for these changes to `Stack`?

File: src/components/toggle-group/ToggleGroupButton.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+const px = {
+  sm: '$4',
+  md: '$5',
+  lg: '$5'
+}
+
+const spacingBetweenElements = {
+  sm: '8px',
```
Comment (hollg) [inline src/components/toggle-group/ToggleGroupButton.tsx:21]: Can these values be expressed in terms of our tokens? (I always forget the px values for each token ...)

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 2 above)
Comment (hollg) [reply src/components/toggle-group/ToggleGroupItem.tsx:6]: Could you expand on what you want to do with this block? E.g. do you want to be able to spread it into _any_ component? Or maybe to create a component that applies the styles to its `children`, which we could wrap around other things?

File: src/components/toggle-group/ToggleGroupItem.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    boxShadow: 'white 0px 0px 0px 2px, $colors$primary 0px 0px 0px 4px'
+  }
+}
+
+export const StyledItem = styled(ToggleGroup.Item, {
+  bg: 'white',
+  color: '$tonal400',
+  border: '1px solid $tonal200',
+  cursor: 'pointer',
+  outline: 'none !important',
```
Comment (hollg) [inline src/components/toggle-group/ToggleGroupItem.tsx:21]: Why do we need `!important` here?

File: src/components/stack/Stack.tsx (hunk 3 above)
Comment (LimeWub) [reply src/components/stack/Stack.tsx:52]: Stack was **way** too specific with alignment and wrapping, that specificity was a hinderance to my use case (and I assume to most use cases). 
I wouldn't be able to use it for doing spacing unless I had more flexibility when it comes to wrapping (I needed no-wrap) and alignment (I needed no alignment so that heights/widths would match depending on orientation).

I put what was used to be the default as the default prop to the component so it should all be safe changes and not break anything not explicitly overriding them.

File: src/components/toggle-group/ToggleGroupButton.tsx (hunk 4 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupButton.tsx:21]: ye - I'll try to figure out which token they match to :S

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 2 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupItem.tsx:6]: I think I would like to be able to store these styles somewhere to import and spread to more focusable block components.

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 5 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupItem.tsx:21]: Why do we not need important? :p Is there a chance we would want to override outline for this ever?

I think it was just a quick way of removing it from the focus-visible state tbh, I don't remember why it's separate to the focus-visible style, I assume it's a WIP leftover. 

In fact I'm considering moving this to the focusIndicatorBlock styles....

Will move and test and lyk

File: src/components/stack/Stack.tsx (hunk 3 above)
Comment (LimeWub) [reply src/components/stack/Stack.tsx:52]: I just realised `column` with `align-items: center` may be a problem. Will test/update this.

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 5 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupItem.tsx:21]: Moving to focusIndicatorBlock w no `!important` has same effect.

File: src/components/stack/Stack.tsx (hunk 3 above)
Comment (LimeWub) [reply src/components/stack/Stack.tsx:52]: ^ Yup there was a change to behaviour introduced. I've added a line to fix this.

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 2 above)
Comment (hollg) [reply src/components/toggle-group/ToggleGroupItem.tsx:6]: Are you struggling with where to store them, or how to import them elsewhere? We have a `utils` directory which I think would make sense as a place to store them — but I would only go down that route atm if you've already spoken to Miguel about the idea of using this elsewhere too.

File: src/components/stack/Stack.tsx (hunk 3 above)
Comment (hollg) [reply src/components/stack/Stack.tsx:52]: I feel a little sceptical about these changes. I think `Stack` does something quite specific _by design_. Did you not just want a `Flex`?

Comment (LimeWub) [reply src/components/stack/Stack.tsx:52]: > Stack is a layout component that provides an abstraction over the flexbox gap property (~design system)

Why would this component have opinions of alignment, wrap and justify? It doesn't make sense that it would have opinions if it's the 'component to do gaps with'.

Tbf I don't even think we should have defaults for align/gap/etc but these are there (and the new ones have been added) just so the components already using them don't break.

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 2 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupItem.tsx:6]: Ye I was struggling about where to store it - I didn't notice the utils directory/wasn't sure if that's where this should go or if there's one just for css utils/stitches - I'm going to move it it in there

Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupItem.tsx]: moved 👍

File: src/components/stack/Stack.tsx (hunk 3 above)
Comment (hollg) [reply src/components/stack/Stack.tsx:52]: > Why would this component have opinions of alignment, wrap and justify? It doesn't make sense that it would have opinions if it's the 'component to do gaps with'.

This is a good question — I'm not sure! But that's what makes me hesitant. I totally see the logic in what you're saying, it just seems to me that there are two possibilities here: either the code is over-reaching for what it's supposed to do and the docs are an accurate description, or the docs are wrong/outdated and `Stack` is intentionally doing more than just providing gaps.

Gonna ping @thomasdigby and @andoulla since I think one of them wrote the first version of `Stack` and maybe they can share their thoughts.

Comment (andoulla) [reply src/components/stack/Stack.tsx:52]: I didn't write it but I remember that intentionality as I struggled to wrap my head around it in the beginning. the intention was to be very specific on what it does as it is just a Flex component with very specific CSS (StackContext follows the same principle). if we make it more flexible then we might as well just use Flex In my opinion.

Comment (LimeWub) [reply src/components/stack/Stack.tsx:52]: The difference with Flex is that it Stack does gap spacing to its children. That's why I use Stack.

File: src/components/stack/Stack.tsx [hunk 6]
```diff
@@ -34,28 +34,35 @@ const gap = Object.keys(theme.space).reduce(
...
-        flexDirection: 'row-reverse',
-        alignItems: 'center'
+        flexDirection: 'row-reverse'
       },
       column: {
         flexDirection: 'column'
       }
     },
+    wrap: {
+      true: {
```
Comment (andoulla) [inline src/components/stack/Stack.tsx:53]: Having flexWrap as a truthy variant might be confusing as for all the other variants  justify = "start" which as develop tells me which option it applies

Comment (LimeWub) [reply src/components/stack/Stack.tsx:53]: Would you prefer?: `wrap = "wrap" | "no-wrap" | "wrap-reverse"`

I did as true/false as it covers "wrap" | "no-wrap" without repeating the variant name in the value.

File: src/components/stack/Stack.tsx (hunk 3 above)
Comment (thomasdigby) [reply src/components/stack/Stack.tsx:52]: I think the phrase "component to do gaps with" isn't really accurate, perhaps the opening line in the documentation isn't as accurate as it could be, but the stack component kinda just does what it says, it allows you to stack things and provide space between them. Usually, this would just be `Flex` with a `gap` property, but we need a few extra layers to replicate that behaviour when it's unsupported.

I don't think "wrap" aligns with the intent of this component, but I can understand why it might be useful. Is this something required of the toggle group?

File: src/components/stack/Stack.tsx (hunk 6 above)
Comment (andoulla) [reply src/components/stack/Stack.tsx:53]: I think I do because it would be more consistent with the other ones. It will be easier to remember

File: src/components/stack/Stack.tsx (hunk 3 above)
Comment (LimeWub) [reply src/components/stack/Stack.tsx:52]: `flexWrap: 'wrap'` and `alignItems: 'center'` were there before - I just picked them out to props so I'm able to use this component without redoing gap logic with flex.

Comment (thomasdigby) [reply src/components/stack/Stack.tsx:52]: Cool, I can get on board with that, although I do think that we ought to consider the overlap between `Flex` and `Stack` components, these changes definitely move them closer together which isn't necessarily a bad thing but we need to document those differences and use-cases for both. (in future)

One thing that we'll need to be careful with is that changing the styles and props for `Stack` will require a major version change, so can we ensure that we set defaults to match how it currently renders.

Comment (LimeWub) [reply src/components/stack/Stack.tsx:52]: Just to make sure I'm understanding this:
I've set the defaults to match the old functionality (will do another check to make sure) so I don't think it's really a breaking change BUT do you want me to mark it as a BREAKING CHANGE anyway and get a major version bump?

Comment (hollg) [reply src/components/stack/Stack.tsx:52]: @LimeWub apologies for my misunderstanding about what changed here/the reasoning earlier! [this comment](https://github.com/Atom-Learning/components/pull/333#discussion_r875609690) made it all clearer to me :D

I definitely agree about avoiding a breaking change but from what I've seen I think your default props should do that. Maybe we could create some snapshots with the old version and then run them again with the new one to check.

I'm gonna resolve this thread 👍🏻

File: src/components/icon/Icon.tsx [hunk 7]
```diff
@@ -3,7 +3,7 @@ import * as React from 'react'
 import { styled } from '~/stitches'
 import { Override } from '~/utilities/types'
 
-const StyledIcon = styled('svg', {
+export const StyledIcon = styled('svg', {
```
Comment (hollg) [inline src/components/icon/Icon.tsx:6]: How confident are we that we can't target the `Icon` in nested styles without exporting `StyledIcon` directly? If it's the only way then fair enough, but it would be really nice if we could avoid it. I really thought we had managed this before 🤔

**Comment by LimeWub** [comment]
> I missed this in my previous review, but we need some docs for the new component 😄

Yikes! I think I had them in there and I resolved/deleted them. Will re-add!

File: src/components/icon/Icon.tsx (hunk 7 above)
Comment (LimeWub) [reply src/components/icon/Icon.tsx:6]: I'll try it again.

**Comment by andoulla** [comment]
Just a suggestion, but I think we should amend the documentation to reflect some of the things we discussed, on the intentions of the component and the new variants. This way in 6months we can avoid having the same confusion

**Comment by LimeWub** [comment]
> I think we should amend the Stack documentation
I 100% agree on this - But is the Stack documentation in a separate repo rn? I think so and I think @hollg is merging it with this one so I will update it when it's merged (or at least that's the plan)

File: src/components/icon/Icon.tsx (hunk 7 above)
Comment (LimeWub) [reply src/components/icon/Icon.tsx:6]: Ye, It doesn't work with `Icon`

**Comment by LimeWub** [comment]
> Just a suggestion, but I think we should amend the Stack documentation to reflect some of the things we discussed, on the intentions of the component and the new variants

@andoulla Ok I thought I'd just update the table at the bottom of the page here: https://design.atomlearning.technology/components/stack But I think that may auto-update? I don't think the text actually needs any edits.

File: src/components/stack/Stack.tsx [hunk 8]
```diff
@@ -67,17 +80,23 @@ export const Stack: React.FC<StackProps> = ({
   css,
   gap = 2,
   direction = 'row',
+  wrap = 'wrap',
   justify = 'start',
+  align,
```
Comment (andoulla) [inline src/components/stack/Stack.tsx:85]: why not add a default to align?

File: src/components/toggle-group/ToggleGroup.mdx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+  <ToggleGroup.Button value="b">B</ToggleGroup.Button>
+  <ToggleGroup.Button value="icon">
+    <Icon is={Upload} />
+  </ToggleGroup.Button>
+</ToggleGroup.Root>
+```
+
+## Full width
+
+`isFullWidth={boolean}`
```
Comment (andoulla) [inline src/components/toggle-group/ToggleGroup.mdx:54]:
```suggestion
`isFullWidth={true|false}`
```

just a small suggestion, as the other examples list the values not the type

File: src/components/toggle-group/ToggleGroup.test.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    <ToggleGroup.Button value="button_b">button B</ToggleGroup.Button>
+    <ToggleGroup.Button value="button_icon">
+      <Icon is={Upload} />
+      Button with icon
+    </ToggleGroup.Button>
+  </ToggleGroup.Root>
+)
+
+describe('ToggleGroup', () => {
+  it('renders', async () => {
```
Comment (andoulla) [inline src/components/toggle-group/ToggleGroup.test.tsx:30]: I would suggest testing the other variants too, especially the disabled one.

File: src/components/toggle-group/ToggleGroupButton.tsx [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+    flexShrink: 0
+  },
+  variants: {
+    size: {
+      sm: getSizeVariant('sm'),
+      md: getSizeVariant('md'),
+      lg: getSizeVariant('lg')
+    },
+    isIconOnly: {
+      true: {
```
Comment (andoulla) [inline src/components/toggle-group/ToggleGroupButton.tsx:53]: not sure if feasible, but it feels that we should be able to pass the isIconOnly into getSizeVariant and just not set margin, instead of setting it and then having to overwrite it

Comment (andoulla) [reply src/components/toggle-group/ToggleGroupButton.tsx:53]: Also, if that's not a choice what if we tried having 
```
 isIconOnly: {
      true: {}
}
```

and in the compountVariant part, we tried to set the py and px in the CSS prop there. Will, that help avoid the important?

File: src/components/toggle-group/ToggleGroupButton.tsx [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+    }
+  ]
+})
+
+export const ToggleGroupButton: React.FC<
+  React.ComponentProps<typeof StyledButton>
+> = ({ size, children, ...rest }) => {
+  const childrenArray = React.Children.toArray(children)
+  const isSingleChild = childrenArray.length <= 1
+  const isIconOnly =
```
Comment (andoulla) [inline src/components/toggle-group/ToggleGroupButton.tsx:89]: I like this, much cleaner than other places

File: src/components/toggle-group/ToggleGroupItem.tsx [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+      borderColor: 'currentColor !important',
+      color: '$primaryMid'
+    },
+    '&:focus-visible': {
+      ...focusVisibleStyleBlock(),
+      '&[data-state="off"]': {
+        borderColor: '$tonal200 !important'
+      },
+      '&[data-state="on"]': {
+        boxShadow: `inset currentColor 0px 0px 0px 1px, ${
```
Comment (andoulla) [inline src/components/toggle-group/ToggleGroupItem.tsx:30]: should this shadow be added into the theme?

File: src/components/toggle-group/ToggleGroupRoot.tsx [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+          '&:not(:last-child)': {
+            borderBottomColor: 'transparent'
+          }
+        }
+      }
+    }
+  ]
+})
+
+const orientationToDirection = (o) => {
```
Comment (andoulla) [inline src/components/toggle-group/ToggleGroupRoot.tsx:113]: Very minor, I always advocate for using the full word instead of the initial as it increases readability. we don't gain anything with shortening it

File: src/components/toggle-group/ToggleGroupRoot.tsx [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+  switch (o) {
+    case 'horizontal':
+      return 'row'
+    case 'vertical':
+      return 'column'
+    default:
```
Comment (andoulla) [inline src/components/toggle-group/ToggleGroupRoot.tsx:119]: I think we don't need the switch as this is a binary check and always will be, it can either be horizontal or vertical. an if statement is more than enough and we don't need to have this empty default

`return orientation === "horizontal" ? "row": column"`

File: src/components/toggle-group/ToggleGroup.mdx [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+component: ToggleGroup
+description: Extends functionality the Toggle Group radix component
+category: Content
+---
+
+Lets users toggle between a group of limited options(Suggested range is 2-6 options).
+
+Functionality based on the [`ToggleGroup`](https://www.radix-ui.com/docs/primitives/components/toggle-group) radix component, which already allows for: single/multiple select, disabling or partly disabling options, adds keyboard navigation and orientation and more.
+
+Extends visually by allowing for vertical/horizontal display, gaps using stack or joined, rounded corners and so on.
```
[the remark this answers is not in the document]
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroup.mdx:12]: Aren't the examples below enough information regarding visual changes? 
I don't see the point of saying I'm adding a white background and primary colour font, I think that's unnecessary info to the user who can just see the examples etc.

File: src/components/toggle-group/ToggleGroup.mdx (hunk 9 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroup.mdx:54]: The other values expect an enum type of value though; I'm not necessarily against this - although boolean is always `true|false` - so I don't see a benefit to this change other than opinion

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 13 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupItem.tsx:30]: I'm not sure I'm understanding this but I don't thinkkkkkkkk soooooo? It's just a visual for this specific component.

File: src/components/toggle-group/ToggleGroupRoot.tsx (hunk 14 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupRoot.tsx:113]: I agree, I'll change it.

File: src/components/toggle-group/ToggleGroupRoot.tsx (hunk 15 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupRoot.tsx:119]: you're right - I'll change this to either an if or the above ternary

File: src/utilities/style/focus-visible-style-block.ts [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+  position: string
+  zIndex: number
+  boxShadow: string
+}
+
+export const focusVisibleStyleBlock = ({
+  position = 'relative',
+  zIndex = 1
+}: focusVisibleStyleBlockPropsType = {}): focusVisibleStyleBlockReturnType => {
+  return {
```
[the remark this answers is not in the document]
Comment (LimeWub) [reply src/utilities/style/focus-visible-style-block.ts:21]: Ye maybe - not sure how I'd go about it but will have a look.

File: src/components/toggle-group/ToggleGroupButton.tsx (hunk 11 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupButton.tsx:53]: 1) Not sure how I can pass a prop from a stitches variant - I think it may not be possible but can have a look.

2) If I had compound variant then I'd need to have 3 variants (1 for each size) which I think is messier than the `!important`. 

I do understand no liking the `!important`, although I don't see us ever wanting to override it....

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 13 above)
Comment (andoulla) [reply src/components/toggle-group/ToggleGroupItem.tsx:30]: oh this is a specific shadow just for this component? 

What I meant is that we do have some default shadows in our theme repo, and I was just wondering if we could either use something from there or add this one there. However, if it is only for this component then we don't need to do it. 

I was just thinking that we were trying to standardise a bit the different shadows we had. (but every rule has exceptions)

File: src/components/toggle-group/ToggleGroupButton.tsx (hunk 11 above)
Comment (andoulla) [reply src/components/toggle-group/ToggleGroupButton.tsx:53]: because I have been caught out saying we will never want to overwrite it and then having to overwrite something, my suggestion is that if we can avoid locking it we should.

I don't see the harm in having 3 compound variants, to be honest. The button for example has quite a lot more

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 13 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupItem.tsx:30]: Oh ye it's actually a 'hack' inside shadow to make the border look like 2px wide when selected rather than 1px - very specific to this design.

File: src/components/stack/Stack.tsx (hunk 8 above)
Comment (andoulla) [reply src/components/stack/Stack.tsx:85]: quick q, is Align mandatory for TS or optional?

File: src/components/toggle-group/ToggleGroup.mdx (hunk 16 above)
[the remark this answers is not in the document]
Comment (andoulla) [reply src/components/toggle-group/ToggleGroup.mdx:12]: oh then the so on is unnecessary in my opinion. to make it more precise I would just switch to something like. I would just list the things the Dev can control with the variants
```suggestion
Extends the Radix component visually by allowing for vertical/horizontal display and gaps using stack or joined.
```

File: src/components/toggle-group/ToggleGroup.mdx (hunk 9 above)
Comment (andoulla) [reply src/components/toggle-group/ToggleGroup.mdx:54]: this is why I said it is a small suggestion.  it is just a consistency thing. not to precious on it though

File: src/components/toggle-group/ToggleGroup.test.tsx (hunk 10 above)
Comment (andoulla) [reply src/components/toggle-group/ToggleGroup.test.tsx:30]: fair enough about the disabled part. what about testing a single option with horizontal orientation.

File: src/components/toggle-group/ToggleGroupItem.tsx (hunk 13 above)
Comment (andoulla) [reply src/components/toggle-group/ToggleGroupItem.tsx:30]: 👍

File: src/components/toggle-group/ToggleGroup.test.tsx (hunk 10 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroup.test.tsx:30]: Orientation is part of ToggleGroup (functional), and direction (visual) is part Stack :o

File: src/components/toggle-group/ToggleGroup.mdx (hunk 9 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroup.mdx:54]: imma leave it as it is :)

File: src/components/toggle-group/ToggleGroup.mdx (hunk 16 above)
[the remark this answers is not in the document]
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroup.mdx:12]: Okay - I'lll reword 👍

File: src/components/stack/Stack.tsx (hunk 8 above)
Comment (LimeWub) [reply src/components/stack/Stack.tsx:85]: I'm not sure - it uses the `typeof` the stitches setup, I'm not getting good typescript highlighting for it tho...

```
type StackProps = React.ComponentProps<typeof StyledStack> & { css?: CSS }
```

Comment (andoulla) [reply src/components/stack/Stack.tsx:85]: because I think it might be mandatory where in that situation users will have to pass it in anyway. We will know as soon as you use it in the feature you need it for right?

Comment (LimeWub) [reply src/components/stack/Stack.tsx:85]: It doesn't suggest me props when I use it in ToggleGroup.Root :S

For what it's worth I pass `align={false}` in my use case for ToggleGroup so that it never aligns and just stretches instead.

File: src/components/toggle-group/ToggleGroupButton.tsx (hunk 11 above)
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroupButton.tsx]: There were actually already compound variants for isIconOnly+size .-. lol
I stuck a clear padding there - much better than the importants, good catch 👍

File: src/components/stack/Stack.tsx (hunk 8 above)
Comment (andoulla) [reply src/components/stack/Stack.tsx:85]: I wonder why it doesn't suggest Props, is it a stitches bug or something wrong in our setup. but I diverge

File: src/components/toggle-group/ToggleGroup.mdx (hunk 16 above)
[the remark this answers is not in the document]
Comment (LimeWub) [reply src/components/toggle-group/ToggleGroup.mdx]: done - think it's cleaner now.

**Review by hollg — APPROVED** [review]

**Review by andoulla — APPROVED** [review]

File: src/utilities/style/focus-visible-style-block.ts (hunk 17 above)
[the remark this answers is not in the document]
Comment (LimeWub) [reply src/utilities/style/focus-visible-style-block.ts:21]: I tried doing this with a Snapshot but couldn't get it to render out the CSS so it wasn't testing right.
I don't know how else to test it.

**code:**
Screenshot 2022-05-20 at 14 05 28

**result:**
Screenshot 2022-05-20 at 14 05 09

If you have any idea what looks wrong with this test lmk :/ planning to merge this today at some point.
